pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -361,7 +361,15 @@ func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType cluster
 	}
 	ec := NewMutableCluster(c)
 	switch discoveryType {
-	case cluster.Cluster_STRICT_DNS, cluster.Cluster_LOGICAL_DNS:
+	case cluster.Cluster_LOGICAL_DNS:
+		if len(localityLbEndpoints) != 1 || len(localityLbEndpoints[0].LbEndpoints) != 1 {
+			log.Warnf("%s cluster of type DNS_ROUND_ROBIN should have a single locality_lb_endpoint and a single lb_endpoint."+
+				" This can happen if two Service Entries are created with same host name, resolution as DNS_ROUND_ROBIN and with"+
+				" different endpoints.", name)
```

Comment:
Here for the issue it is the  same endpoint

Reply:
it does not matter whether it is same or different. LOGICAL_DNS can not have more than one

https://github.com/envoyproxy/envoy/blob/a4053922d7311e11c18e764ff5e5bfa8f71cd334/source/common/upstream/logical_dns_cluster.cc#L71

Reply:
Yes, i mean the desc could be including the case